streaming/src/test/scala/org/apache/spark/streaming/ui/StreamingJobProgressListenerSuite.scala

```diff
@@ -116,4 +171,73 @@ class StreamingJobProgressListenerSuite extends TestSuiteBase with Matchers {
     listener.retainedCompletedBatches.size should be (limit)
     listener.numTotalCompletedBatches should be(limit + 10)
   }
```

Comment:
Yes. The limitation of the `batchTimeToOutputOpIdSparkJobIdPair`'s size is at the end of `test("out-of-order onJobStart and onBatchXXX")`.